hackDetail.js

```diff
@@ -13,15 +13,15 @@ export const detail = async (event, context) => {
     });
     conn.model(
       "Hack",
-      new mongoose.Schema({ title: String, description: String, goal: String, team: Array })
+      new mongoose.Schema({ title: String, description: String, goal: String, creator: String, team: Array })
     );
     conn.model("User", new mongoose.Schema({ name: String, email: String }));
   }
   const Hack = conn.model("Hack");
 
   try {
 
-    let result = await Hack.findById(id).populate('team', '-email', 'User');
+    let result = await Hack.findById(id).populate('team', '-email', 'User').populate('creator', '-email', 'User');
```

Comment:
just been reading about using [`ref`](https://mongoosejs.com/docs/api.html#schematype_SchemaType-ref0 in mongoose to do this population of user for you and `select` to auto redact the email when querying.

```

conn.model("Hack", new mongoose.Schema({ 
  title: String, 
  description: String, 
  goal: String, 
  creator: { type: ObjectId, ref: 'User' }, 
  team: { type: [ObjectId], ref: 'User'}
}));

conn.model("User", new mongoose.Schema({
  name: String, 
  email: { type: String, select: false }
}));

...

let result = await Hack.findById(id).populate('team').populate('creator')
```

Not sure if it will work... not tested... but might be worth trying to clean up some of this code.